core/tests/puppeteer-acceptance-tests/utilities/user/topic-manager.ts

```
@@ -306,6 +319,181 @@ export class TopicManager extends BaseUser {
     );
     expect(actualMessage).toEqual(expectedMessage);
   }
+
+  /**
+   * Assign a skill to a topic.
+   *
+   * @param {string} topicName - The name of the topic to assign the skill to.
+   * @param {string} skillName - The name of the skill to assign.
+   */
+  async assignSkillToTopic(
+    skillName: string,
+    topicName: string
+  ): Promise<void> {
+    const isMobileWidth = this.isViewportAtMobileWidth();
+    const skillOptions = isMobileWidth ? mobileSkillsOption : skillEditBox;
+    const assignSkillButton = isMobileWidth
+      ? assignSkillButtonMobile
+      : assignSkillButtonDesktop;
+
+    await this.navigateToTopicAndSkillsDashboardPage();
+    await this.navigateToSkillTab();
+
+    const skillItem = await this.selectSkill(skillName);
+
+    const skillOptionsElement = await skillItem.$(skillOptions);
+    if (skillOptionsElement) {
+      await skillOptionsElement.click();
```

What if skillOptionsElement is not present? should we throw an error for that?

---

Done.